torch/_inductor/ir.py

```diff
@@ -4542,6 +4544,52 @@ def __init__(
         self.kwargs = {} if kwargs is None else kwargs
         V.graph.warn_fallback(self.python_kernel_name)
 
+        # args that are aliased
+        self.alias_names = []
+        # args that are mutated AND returned from the op
+        self.mutation_names = []
+
+        if isinstance(self.op_overload, torch._ops.HigherOrderOperator):
+            # We assume here that HOPs with FallbackKernel are functional.
+            # This may not always be true! HOPs must individually opt-in to
+            # FallbackKernel, so please check this if you opt-in.
+            return
+
+        schema = self.op_overload._schema
+
+        # NOTE: [FallbackKernel supported operators]
+        # We only support three types of operators:
+        # - functional ops
+        # - view ops
+        # - inplace aten ops
+        # - mutating ops that are auto-functionalizable. That is,
+        # the operator may mutate any number of inputs, but its outputs
+        # may not alias any of the inputs.
+        #
+        # The unsupported cases usually do not show up here (because
+        # AOTAutograd functionalized them away); the only way for an in-place
+        # op to show up here is if a lowering or pass introduced it.
+        if torch._library.utils.is_inplace_aten_op(self.op_overload):
+            self.mutation_names.append(tensor_args[0].get_name())
+            return
+
+        if schema.is_mutable and not can_auto_functionalize(kernel):
+            raise NotImplementedError(
```

Comment:
Why cant we handle this case ?

Reply:
For custom ops, the schema is not enough information to tell us about the semantics of the operator. For example, if the operator has a schema that looks like "(Tensor(a!) x) -> Tensor(a)", it's unclear if the operator is (1) returning x as-is or (2) returning a view on x.

For aten operators, we have the convention that it is always (1). It's unclear to me how to generate code when we don't know which of these two situations we're in.

Pragmatically, I've never seen a custom op have a schema that looks like the above, so, this is not a case we need to worry about.